python/lsst/analysis/tools/atools/diffMatched.py

```diff
@@ -231,25 +422,31 @@ def configureMetrics(
         key_flux = self.config_mag_x.key_flux
 
         units = {}
-        for name, name_class in zip(self._names, self._types):
-            name_capital = name.capitalize()
+
+        for object_class in self.get_classes():
+            name_type = self.get_class_type(object_class)
+            name_type_plural = self.get_class_name_plural(object_class)
+            name_capital = name_type_plural.capitalize()
             x_key = f"x{name_capital}"
 
-            for minimum in range(self._mag_low_min, self._mag_low_max + 1):
-                action = getattr(self.process.calculateActions, f"{name}{minimum}{attr_suffix}")
+            # Set up metrics for objects of one class within a magnitude range
+            bins = self.mag_bins.get_bins()
+            for minimum in bins:
+                action = getattr(self.process.calculateActions, f"{name_type_plural}_{minimum}{attr_suffix}")
                 action.selector_range = RangeSelector(
                     vectorKey=x_key,
-                    minimum=minimum,
-                    maximum=minimum + self._mag_interval,
+                    minimum=minimum / 1000.0,
```

Comment:
It seems particularly silly for these to be defined in millimags and then /1000 everywhere.

Reply:
We do tend use mmag on the y axis.